mlxtend/feature_selection/exhaustive_feature_selector.py

```diff
@@ -227,34 +269,39 @@ def fit(self, X, y, custom_feature_names=None, groups=None, **fit_params):
         else:
             X_ = X
 
+        # shouldn't we use X_ now?
         if custom_feature_names is not None and len(custom_feature_names) != X.shape[1]:
             raise ValueError(
                 "If custom_feature_names is not None, "
                 "the number of elements in custom_feature_names "
                 "must equal the number of columns in X."
             )
 
+        if self.feature_groups is None:
+            self.feature_groups = [[i] for i in range(X_.shape[1])]  # use X?
```

Comment:
Move this if-block to line 222, just before `self.feature_groups = feature_groups`